app/models/duration.rb

```diff
@@ -50,87 +52,58 @@ def format_casual(num_units: 2, sign: :negatives)
     d = d.first(num_units).to_h.map { |k, v| "#{v.to_i}#{k}" }.join(' ')
     return "+#{d}" if sign == :always && positive?
     return "-#{d}" if [:always, :negatives].include?(sign) && negative?
-    d
-  end
-
-  def ==(duration)
-    # Normally we'd call Duration#nil? here, but it treats 0 as nil to deal with old durations in the db where 0 means
-    # absent. When those are cleaned up, we can change Duration#nil? and this.
-    return false if duration == nil || to_ms.nil? || (duration.respond_to?(:to_ms) && duration.to_ms.nil?)
-    return false unless duration.respond_to?(:to_ms)
-
-    to_ms == duration.to_ms
-  end
-
-  def !=(duration)
-    return false if nil? || duration.nil?
-    return false unless duration.respond_to?(:to_ms)
-
-    to_ms != duration.to_ms
-  end
-
-  def <(duration)
-    # Normally we'd call Duration#nil? here, but it treats 0 as nil to deal with old durations in the db where 0 means
-    # absent. When those are cleaned up, we can change Duration#nil? and this.
-    return false if duration == nil || to_ms.nil? || (duration.respond_to?(:to_ms) && duration.to_ms.nil?)
 
-    # duration can be a Duration or a number of milliseconds
-    ms = duration.respond_to?(:to_ms) ? duration.to_ms : duration
-
-    to_ms < ms
-  end
-
-  def <=(duration)
-    self < duration || self == duration
-  end
-
-  def >(duration)
-    # Normally we'd call Duration#nil? here, but it treats 0 as nil to deal with old durations in the db where 0 means
-    # absent. When those are cleaned up, we can change Duration#nil? and this.
-    return false if duration == nil || to_ms.nil? || (duration.respond_to?(:to_ms) && duration.to_ms.nil?)
-
-    # duration can be a Duration or a number of milliseconds
-    ms = duration.respond_to?(:to_ms) ? duration.to_ms : duration
-
-    to_ms > ms
+    d
   end
 
-  def >=(duration)
-    self > duration || self == duration
+  def !=(other)
+    !to_ms == other.to_ms
```

Comment:
Is this right? Feels like `!to_ms` would always return `true` or `false`.

Reply:
Nope!  I meant to put `!(self == other)` and leverage the automatically set up `==` method since we know the reverse is just as simple as normal equality.  I'm not sure why it's not covered by Comparable to be honest.